synfig-core/src/synfig/rendering/surface.cpp

```diff
@@ -184,6 +184,20 @@ Surface::get_pixels(Color *dest) const
 	return get_pixels_vfunc(dest);
 }
 
+bool
+Surface::compare_with(synfig::rendering::Surface::Handle s) const
+{
+	if(!(get_width()==s->get_width() && get_height()==s->get_height())){
+		return false;
+	}else{
+		std::vector<Color> buffer1(get_pixels_count()),buffer2(s->get_pixels_count());
+		get_pixels(&buffer1.front());
```

Comment:
I'm not sure if it should be this way.
`get_pixels()` returns a pointer to an C-style array. so you should use `memcmp`

Reply:
Okay! Will do!